services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/services/services_tracker/_tracker.py

```diff
@@ -0,0 +1,85 @@
+import json
+import logging
+from asyncio import Task
+from typing import Final
+
+from fastapi import FastAPI
+from models_library.projects_nodes_io import NodeID
+from pydantic import NonNegativeFloat, NonNegativeInt
+from servicelib.background_task import start_periodic_task, stop_periodic_task
+from servicelib.redis import RedisClientSDK
+from servicelib.redis_utils import exclusive
+from servicelib.utils import logged_gather
+
+from ._notifier import publish_message
+from ._resource_manager import ServicesManager
+from ._status_cache import ServiceStatusCache
+
+_DIRECTOR_V2_API_PARALLELISM: Final[NonNegativeInt] = 10
+
+_logger = logging.getLogger(__name__)
+
+
+class ServicesTracker(ServicesManager):
+    def __init__(
+        self,
+        app: FastAPI,
+        redis_client_sdk: RedisClientSDK,
+        check_interval: NonNegativeFloat = 10,
+    ) -> None:
+        super().__init__(app, redis_client_sdk)
+
+        self.check_interval = check_interval
+        self.service_status_cache: ServiceStatusCache = ServiceStatusCache(
+            app, ttl=self.check_interval * 4, namespace="services_tracker"
+        )
+
+        self._check_task: Task | None = None
+
+    async def setup(self) -> None:
+        await super().setup()
+
+        lock_key = f"lock:{self.class_path()}:services-tracker"
+        lock_value = json.dumps({})
+        self._check_task = start_periodic_task(
+            exclusive(self._redis_client_sdk, lock_key=lock_key, lock_value=lock_value)(
+                self._check_services_status_task
+            ),
+            interval=self.cleanup_interval,
```

Comment:
so you use that pattern here again. that means only one instance of dynamic scheduler can run the task at a time. but that does not prevent another of running it too.
Now it depends on what you wanna do...